assert/assert_is_error.ts

```diff
@@ -23,11 +23,11 @@ export function assertIsError<E extends Error = Error>(
   error: unknown,
   // deno-lint-ignore no-explicit-any
   ErrorClass?: new (...args: any[]) => E,
-  msgIncludes?: string,
+  msgIncludes?: string | RegExp,
```

Comment:
This is a change to the public API, however it only adds a new type and should not break any backwards compatibility.

Reply:
This change makes sense to me. If we do this type change, I think the parameter name `msgMatches` would be more appropriate. What do you think?